src/outputs/parthenon_xdmf.cpp

```diff
@@ -150,26 +153,23 @@ void genXDMF(std::string hdfFile, Mesh *pm, SimTime *tm, int nx1, int nx2, int n
     // write graphics variables
     int ndim;
     for (const auto &vinfo : var_list) {
-      std::vector<hsize_t> alldims(
-          {static_cast<hsize_t>(vinfo.nx6), static_cast<hsize_t>(vinfo.nx5),
-           static_cast<hsize_t>(vinfo.nx4), static_cast<hsize_t>(vinfo.nx3),
-           static_cast<hsize_t>(vinfo.nx2), static_cast<hsize_t>(vinfo.nx1)});
-      // Only cell-based data currently supported for visualization
-      if (vinfo.where == MetadataFlag(Metadata::Cell)) {
-        ndim = 3 + vinfo.tensor_rank + 1;
-        for (int i = 0; i < vinfo.tensor_rank; i++) {
-          dims[1 + i] = alldims[3 - vinfo.tensor_rank + i];
-        }
-        dims[vinfo.tensor_rank + 1] = nx3;
-        dims[vinfo.tensor_rank + 2] = nx2;
-        dims[vinfo.tensor_rank + 3] = nx1;
-      } else {
+      // JMM: I can't figure out how to get faces/edges to work and
+      // I'm not going try any longer. More eyes appreciated.
+      if ((vinfo.where != MetadataFlag({Metadata::Cell})) &&
+          (vinfo.where != MetadataFlag({Metadata::Node}))) {
         continue;
       }
-
+      ndim = vinfo.FillShape<hsize_t>(domain, &(dims[1])) + 1;
       const int num_components = vinfo.num_components;
+      nx3 = dims[ndim - 3];
+      nx2 = dims[ndim - 2];
+      nx1 = dims[ndim - 1];
+      dims321 =
+          std::to_string(nx3) + " " + std::to_string(nx2) + " " + std::to_string(nx1);
```

Comment:
How does this interact with the `dims321` set above? Or in other words, is this required here or a leftover from the non-cell centered field attempts?

Reply:
`dims321` above can be deleted. It's only relevant here.